compiler/rustc_middle/src/ty/destructured_const.rs

```diff
@@ -0,0 +1,8 @@
+use crate::ty::{self, VariantIdx};
+
+/// The constituent parts of a type level constant of kind ADT or array.
+#[derive(Copy, Clone, Debug, HashStable)]
+pub struct DestructuredConst<'tcx> {
```

Comment:
This is only used in pretty printing and symbol mangling... I wonder if we can get rid of it.